plugin/pkg/scheduler/factory/factory.go

```diff
@@ -227,9 +237,7 @@ func (f *ConfigFactory) CreateFromKeys(predicateKeys, priorityKeys sets.String,
 		Algorithm:  algo,
 		Binder:     &binder{f.Client},
 		NextPod: func() *api.Pod {
-			pod := f.PodQueue.Pop().(*api.Pod)
-			glog.V(2).Infof("About to try and schedule pod %v", pod.Name)
-			return pod
+			return f.getNextPod()
```

Comment:
good point -- I assume you mean this line and the previous and next can collapse to just
NextPod: f.getNextPod(),